src/node/interfaces.cpp

```diff
@@ -814,14 +814,32 @@ class ChainImpl : public Chain
         });
         return result;
     }
-    bool updateRwSetting(const std::string& name, const common::SettingsValue& value, bool write) override
+    bool updateRwSetting(const std::string& name,
+                         const interfaces::SettingsUpdate& update_settings_func) override
     {
+        std::optional<interfaces::SettingsAction> action;
         args().LockSettings([&](common::Settings& settings) {
-            if (value.isNull()) {
-                settings.rw_settings.erase(name);
-            } else {
-                settings.rw_settings[name] = value;
-            }
+            auto* ptr_value = common::FindKey(settings.rw_settings, name);
+            // Create value if it doesn't exist
+            auto& value = ptr_value ? *ptr_value : settings.rw_settings[name];
```

Comment:
I think this should be simplified to:

<details>
<summary>git diff on 1b41d45d46</summary>

```diff
diff --git a/src/node/interfaces.cpp b/src/node/interfaces.cpp
index 54b986c926..b1699bc7ae 100644
--- a/src/node/interfaces.cpp
+++ b/src/node/interfaces.cpp
@@ -819,10 +819,8 @@ public:
     {
         std::optional<interfaces::SettingsAction> action;
         args().LockSettings([&](common::Settings& settings) {
-            auto* ptr_value = common::FindKey(settings.rw_settings, name);
-            // Create value if it doesn't exist
-            auto& value = ptr_value ? *ptr_value : settings.rw_settings[name];
-            action = update_settings_func(value);
+            auto& existing_value{settings.rw_settings[name]};
+            action = update_settings_func(existing_value);
         });
         if (!action) return false;
         // Now dump value to disk if requested

```
</details>

Reply:
Agree this would be an improvement. But I think it would be better to change behavior and avoid writing null settings (for reasons in other comment). Would probably write as:

```c++
if (auto* value = common::FindKey(settings.rw_settings, name)) {
    action = update_settings_func(*value);
    if (value->isNull()) settings.rw_settings.erase(name);
} else {
    UniValue value;
    action = update_settings_func(*value);
    if (!value.isNull()) settings.rw_settings[name] = std::move(value);
}
```

Reply:
Taken in #30828